### ISSUE TITLE
CRMPlugin Registration Error - Organization Name can not be null or empty

### ISSUE THREAD

**Issue by nickgsoft** [WaelHamze/dyn365-ce-vsts-tasks#123]
We are using DevOps task for "MSCRM Plugin Registration Tool and MSCRM Tool Installer" tasks on Azure Devops and the deployment server is on-premise version 9. we did provided connection string with org name like below, where test is Organization:
<add name="MyCRMServer" connectionString="AuthType=AD;Url=http://contoso:8080/Test; Domain=CONTOSO; Username=jsmith; Password=passcode" />

While running release pipe line i am getting error Organization cannot be null or epty parameter name "organization name". 

Any help why do i get this errror even if the connection string have the org? is there any other field that i needs to provide org information? please assist. I am using preview version 11* of Plugin Registration .

**Comment by WaelHamze** [comment]
@nickgsoft does the ping task work for you? You normally get this error if connecting string is not valid or some other issue with connectivity from agent to CRM.

**Comment by WaelHamze** [comment]
Closing as no activity for a month, please re-open if you are still having issues

**Comment by nickgsoft** [comment]
@WaelHamze  Sorry for delay on this, but can you please provide details on Ping task? How do you set up that?
As far as the server goes, i can remote login from remote desktop and it is live server.

**Comment by nickgsoft** [comment]
ok, i used one of the ping task from your CRM tool set as release task and got below same results:

2020-01-08T17:17:53.7046690Z ##[section]Starting: MSCRM CRM Ping
2020-01-08T17:17:53.7134985Z ==============================================================================
2020-01-08T17:17:53.7135046Z Task         : MSCRM Ping
2020-01-08T17:17:53.7135072Z Description  : A sample task that checks connectivity to a Dynamics 365 environment
2020-01-08T17:17:53.7135536Z Version      : 10.0.6
2020-01-08T17:17:53.7135559Z Author       : Wael Hamze
2020-01-08T17:17:53.7135597Z Help         : Performs a WhoAmIRequest against the target CRM environment
2020-01-08T17:17:53.7135620Z ==============================================================================
2020-01-08T17:17:54.5829418Z ##[error]Organization cannot be null or empty.
Parameter name: Organization Name
2020-01-08T17:17:54.6030049Z ##[section]Finishing: MSCRM CRM Ping

**Comment by WaelHamze** [comment]
@nickgsoft connection string should be in this format: https://docs.microsoft.com/en-us/dynamics365/customerengagement/on-premises/developer/xrm-tooling/use-connection-strings-xrm-tooling-connect

I presume for on-premise it would be:
AuthType=AD;Url=http://contoso:8080/Test;Domain=CONTOSO;Username=jsmith;Password=passcode"

Where are you running the release from? I mean where is the agent hosted?

Please ensure this agent can see the CRM environment (i.e. there is network connectivity between the two)

If you are using private agents, you may need to ensure there agent is setup with the correct proxy settings.

**Comment by nickgsoft** [comment]
- I verified that CRM instance can be accessible from same server where the deployment agent is hosted because agents and CRM servers are both hosted on the same server. 

- I accessed using the browser which works as well. 

- I also  verified the same connection string parameters are on the Plugin registration tool and that tool can connect it to as well. I verified connection string and try few different options based on chat msdn article, still no luck.

My connection string for ping task is below:
<add name="MSCRM" connectionString="Server=win-lo671e7h9pg.devs.local,organization=SSCOCRMDEVVMDB,AuthType=AD;URL=http://win-lo671e7h9pg.devs.local/SSCOCRMDEVVMDB/; Domain=devs.local; Username=crmadmin; Password=*****" />

**Comment by nickgsoft** [comment]
Is there a way to troubleshoot? i see the ping task behind the picture executed powershell script. So is there any handy powershell script that i can try executing on server and see if that works or not and get the error detials?

**Comment by nickgsoft** [comment]
@WaelHamze let me know if you need more details.

**Comment by WaelHamze** [comment]
@nickgsoft 

Yes it does execute this PS: https://github.com/WaelHamze/xrm-ci-framework/blob/master/MSDYNV9/Xrm.Framework.CI/Xrm.Framework.CI.PowerShell.Scripts/Ping.ps1

You can download the module from: https://www.powershellgallery.com/packages/Xrm.Framework.CI.PowerShell.Cmdlets/9.0.0.8

I still think it is a connection string/network/proxy issue.

Does this connection string work using the xrmtoolbox?

One test you can do is fire up the xrmtoolbox or run the script on the agent server. Make sure you run this using the account that the agent is using.

Also ensure you have correct proxy: https://docs.microsoft.com/en-us/azure/devops/pipelines/agents/proxy?view=azure-devops&tabs=windows

If CRM doesn't work via a proxy you may need to add it to the bypass list as per the link above.

Hope this will help to narrow it down.

**Comment by kaufmd** [comment]
I am facing a similar issue. I want to export a solution from the DEV system and import it into another Sandbox system (aka the next stage). I can connect to the DEV system without issues. But when I try to connect to the destination system I get the exact same error message as the OP.
My connection string looks like this: AuthType=Office365;Username=$(CrmDestinationUser);Password=$(CrmDestinationPassword);Url=$(CrmDestinationUrl)

Where
$(CrmDestinationUser) = [user@<domain>.onmicrosoft.com]
$(CrmDestinationPassword) = ***
$(CrmDestinationUrl) = https://<domain>.crm4.dynamics.com

The weird part is that the exact same connection string works for the source system. The only difference between the two is that the destination system has only recently been set up whereas the DEV system has been used for a good 1.5 years already. Any help is highly appreciated.

**Comment by kaufmd** [comment]
> The weird part is that the exact same connection string works for the source system. The only difference between the two is that the destination system has only recently been set up whereas the DEV system has been used for a good 1.5 years already. Any help is highly appreciated.

And after firing up a VM in Azure and running the connection string through XrmToolbox I can honestly say: often the problem is sitting in front of the screen...

Finding #1: always put the password in quotes in the connection string. That way it doesn't matter if there's a semi-colon (;) present in the password.

Finding #2: if you keep getting the error `Invalid Login Information : An unsecured or incorrectly secured fault was received from the other party. See the inner FaultException for the fault code and detail.
An unsecured or incorrectly secured fault was received from the other party. See the inner FaultException for the fault code and detail. => Authentication FailureUnable to Login to Dynamics CRM` consider that your tenant might have conditional access configured. That was the case here for me. And the source system connection worked because the user for this system has an exception from the conditional access restriction. The user for the destination system doesn't.

Just to help some lost soul that might stumble upon the same errors I did.
